crates/payload/primitives/src/lib.rs

```diff
@@ -341,6 +341,17 @@ pub enum EngineApiMessageVersion {
     V4,
 }
 
+/// Determines how we should choose the payload to return.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
+pub enum PayloadKind {
+    /// Returns the best available payload, without waiting for pending job to finish.
+    #[default]
+    BestAvailable,
```

Comment:
this isn't quite right, from CL pov we want a payload as fast as possible, so where this would be relevant is if there's no best payload yet we start racing the empty payload against the in progress payload:

https://github.com/paradigmxyz/reth/blob/b77265e61b06cb6402ac210a01396ebf57c55418/crates/payload/basic/src/lib.rs#L545-L546

so this mode is more like `Earliest`